app/controllers/cart_controller.rb

```diff
@@ -23,6 +23,46 @@ def remove_item
     session[:cart].delete(params[:item_id])
     redirect_to '/cart'
   end
+#this is the beginning of a refactor of the update method
+  # def update
+  #   require "pry"; binding.pry
+  #   case params[:add]
+  #   when true
+  #     if item.inventory_has_reached_limit?(cart)
+  #       flash[:error] = "You've reached the end of this merchant's inventory"
+  #       redirect_to '/cart'
+  #     else
+  #       cart.add_item(item.id.to_s)
+  #       redirect_to '/cart'
+  #     end
+  #   else
+  #
+  #   end
+  #
+  #   redirect_to "/cart"
+  #
+  #
+  # end
+  def update
+    item = Item.find(params[:item_id])
+    if params[:add] == "true"
+      if item.inventory_has_reached_limit?(cart, item)
+        flash[:error] = "You've reached the end of this merchant's inventory"
+        redirect_to '/cart'
+      else
+        cart.add_item(item.id.to_s)
+        redirect_to '/cart'
+      end
 
+    else params[:add] == "false"
+      if cart.contents[item.id.to_s] == 1
+        session[:cart].delete(params[:item_id])
+        redirect_to '/cart'
+      else
+        cart.remove_item(item.id.to_s)
+        redirect_to "/cart"
```

Comment:
You can move the `redirect_to` to the end of the method and remove it from each condition!

Reply:
I might need a short explanation of this method just to understand it better, bus looks great!

Reply:
For sure!  This is the cart_controller update method.  I can review it with you.  Basically, it's making sure that if I want to add a single item from my cart show page, I don't add more then there is inventory, and if I want to take items off, if I reach zero the item is removed from my cart page altogether.  It's nuanced, but it was what the two stories wanted.